dmipy/core/modeling_framework.py

```diff
@@ -2191,6 +2193,383 @@ def __call__(self, acquisition_scheme, **kwargs):
         return E
 
 
+class MultiCompartmentAMICOModel(MultiCompartmentModelProperties):
+    def __init__(self, models, S0_tissue_responses=None, parameter_links=None):
+        self.models = models
+        self.N_models = len(models)
+        if S0_tissue_responses is not None:
+            if len(S0_tissue_responses) != self.N_models:
+                msg = 'Number of S0_tissue responses {} must be same as '\
+                      'number of input models {}.'
+                raise ValueError(
+                    msg.format(len(S0_tissue_responses), self.N_models))
+        self.S0_tissue_responses = S0_tissue_responses
+        self.parameter_links = parameter_links
+        if parameter_links is None:
+            self.parameter_links = []
+
+        self._prepare_parameters()
+        self._prepare_partial_volumes()
+        self._prepare_parameter_links()
+        self._prepare_model_properties()
+        self._check_for_double_model_class_instances()
+        self._prepare_parameters_to_optimize()
+        self._check_for_NMR_and_other_models()
+        self.x0_parameters = {}
+        self._forward_model_matrix = None
+        self._parameter_indices = {}
+
+        if not have_numba:
+            msg = "We highly recommend installing numba for faster function "
+            msg += "execution and model fitting."
+            print(msg)
+        if not have_pathos:
+            msg = "We highly recommend installing pathos to take advantage of "
+            msg += "multicore processing."
+            print(msg)
+
+    def _check_for_NMR_and_other_models(self):
+        model_types = [model._model_type for model in self.models]
+        if "NMRModel" in model_types:
+            if len(np.unique(model_types)) > 1:
+                msg = "Cannot combine 1D-NMR and other 3D model types together"
+                msg += " into a MultiCompartmentAMICOModel."
+                raise ValueError(msg)
+
+    def _check_if_model_orientations_are_fixed(self):
+        # TODO: return True if the orientations are fixed, False o/wise.
```

Comment:
for the other checks I just raise the error inside these internal `_check` function, see the function directly above.

This allows me to keep the `model.fit()` function cleaner, or we can change this if you have a good argument for it.

Reply:
Noted. In the future I think it could make sense to make it boolean in order to check when we have to set the orientations automatically.